src/mlnet/CodeGenerator/CodeGenerationHelper.cs

```diff
@@ -51,7 +52,9 @@ public void GenerateCode()
                 {
                     inputColumnInformation.IgnoredColumnNames.Add(value);
                 }
+                var inferColumnsStopwatch = Stopwatch.StartNew();
                 columnInference = automlEngine.InferColumns(context, inputColumnInformation);
+                InferColumnsEvent.TrackEvent(columnInference.ColumnInformation, inferColumnsStopwatch.Elapsed);
```

Comment:
Let's say we have a per-iteration event, and on-start event, and on-exit event. If the user does 100 iterations, that's 102 log statements. If we increase from 102 to the current 104, from a network round trip perspective, I think we're in the same boat. However, from a telemetry / logging / style perspective, it could be cleaner to have a separate events for InferColumns & ApplicationStart / etc.

Reply:
I realize that this is also laid out more here https://github.com/dotnet/machinelearning-automl/issues/417